easybuild/framework/easyblock.py

```diff
@@ -2898,7 +2901,21 @@ def build_and_install_one(ecdict, init_env):
                 app.cfg.dump(reprod_spec)
                 _log.info("Dumped fully processed easyconfig to %s", reprod_spec)
             except NotImplementedError as err:
-                _log.warn("Unable to dumped fully processed easyconfig to %s: %s", reprod_spec, err)
+                _log.warn("Unable to dump fully processed easyconfig to %s: %s", reprod_spec, err)
+            # also archive the relevant easyblocks
+            reprod_spec = os.path.join(new_log_dir, 'reprod', 'easyblocks')
+            for easyblock_class in inspect.getmro(app_class):
+                easyblock_path = inspect.getsourcefile(easyblock_class)
+                easyblock_basedir, easyblock_filename = os.path.split(easyblock_path)
+                # if we reach the framework modules we are done
+                if easyblock_filename in FRAMEWORK_EASYBLOCK_MODULES:
+                    break
+                else:
+                    # Check if the easyblock should be in the generic subdir
+                    if os.path.basename(easyblock_basedir) == 'generic':
```

Comment:
@ocaisa We have `'generic'` hardcoded in a number of places, it's time we introduce a constant `GENERIC` for this to avoid silly bugs due to typos?

Reply:
Actually the behaviour of including easyblocks has changed I think, I don't think this check is actually necessary (EB automatically puts an included easyblock in the appropriate directory)

Reply:
This is for backward compatiblity... If you include an easyblock with `--include-easyblocks generic/configuremake.py`, it should still work like it did before.

It's true that `--include-easyblocks configuremake.py` will works as expected though, but that doesn't mean we should break people's configurations where they put generic easyblocks in a `generic` subdirectory because that's what EasyBuild required up to some point...

We can deprecate this block of logic though, and make it clear it's no longer required to do this. But on the other hand, people may *like* having the generic easyblock separate from the rest, so it also makes sense to keep supporting this.

Reply:
Hmm, from our configuration I noticed that we now have `--include-easyblocks=$CUSTOM/*.py,$CUSTOM/generic/*.py`. Does `--include-easyblocks=$CUSTOM/*` still work? I'd like to make the reproducability command simple and generic:
```
eb --include-easyblocks=.../reprod/* .../reprod/*.eb
```